hbase/src/main/java/org/apache/metamodel/hbase/HBaseUpdateCallback.java

```diff
@@ -0,0 +1,119 @@
+ */
+public class HBaseUpdateCallback extends AbstractUpdateCallback implements UpdateCallback {
+
+    private final HBaseClient _hBaseClient;
+
+    public HBaseUpdateCallback(final HBaseDataContext dataContext) {
+        super(dataContext);
+        _hBaseClient = new HBaseClient(dataContext.getConnection());
+    }
+
+    @Override
+    public TableCreationBuilder createTable(final Schema schema, final String name) {
+        return new HBaseCreateTableBuilder(this, schema, name);
+    }
+
+    /**
+     * Initiates the building of a table creation operation.
+     * @param schema the schema to create the table in
+     * @param name the name of the new table
+     * @param columnFamilies the columnFamilies of the new table
+     * @return {@link HBaseCreateTableBuilder}
+     */
+    public HBaseCreateTableBuilder createTable(final Schema schema, final String name,
+            final Set<String> columnFamilies) {
+        return new HBaseCreateTableBuilder(this, schema, name, columnFamilies);
+    }
+
+    @Override
+    public boolean isDropTableSupported() {
+        return true;
+    }
+
+    @Override
+    public TableDropBuilder dropTable(final Table table) {
+        return new HBaseTableDropBuilder(table, this);
+    }
+
+    /**
+     * @throws IllegalArgumentException when table isn't a {@link HBaseTable}
+     */
+    @Override
+    public RowInsertionBuilder insertInto(final Table table) {
+        throw new UnsupportedOperationException(
+                "We need an explicit list of columns when inserting into an HBase table.");
+    }
+
+    /**
+     * Initiates the building of a row insertion operation.
+     * @param table Table to get inserts.
+     * @param columns List of {@link HBaseColumn} to insert on.
+     * @return {@link HBaseRowInsertionBuilder}
+     * @throws IllegalArgumentException The table must be an {@link HBaseTable} and the columns list can't be null or empty
+     */
+    public HBaseRowInsertionBuilder insertInto(final Table table, final List<HBaseColumn> columns) {
+        if (columns == null || columns.isEmpty()) {
+            throw new IllegalArgumentException("The hbaseColumns list is null or empty");
+        }
+        if (table instanceof HBaseTable) {
+            return new HBaseRowInsertionBuilder(this, (HBaseTable) table, columns);
+        } else {
+            throw new IllegalArgumentException("Not an HBase table: " + table);
+        }
+    }
+
+    @Override
+    public boolean isDeleteSupported() {
+        return true;
+    }
+
+    /**
+     * @throws IllegalArgumentException when table isn't a {@link HBaseTable}
+     */
+    @Override
+    public RowDeletionBuilder deleteFrom(final Table table) {
+        if (table instanceof HBaseTable) {
+            return new HBaseRowDeletionBuilder(_hBaseClient, table);
+        } else {
+            throw new IllegalArgumentException("Not an HBase table: " + table);
+        }
+    }
+
+    public HBaseClient getHBaseClient() {
```

Comment:
I think access to this method should be `package private`, because we don't want to expose the HBaseClient as part of the MetaModel api.